egs/wsj/s5/steps/nnet3/fvector/generate_variable_length_range.py

```diff
@@ -0,0 +1,188 @@
+#!/usr/bin/env python
+
+# The function use to generate range file for fvector
+# This is the variable-length version
+# The format is <wav_t_start> <wav_t_end> <noise_uttid> <noise_t_start> <noise_t_end> <snr>
+
+# For <wav_t_start> <wav_t_end>
```

Comment:
You can add option --variable-len-additive-noise and then combine two functions. 
You can have random SNR for both of them!